5733d9e2be6485d52ffa08870cabdee0/sandbox-ui
# MGDOBR-872: Manage masked values inside action/source configuration

JIRA: https://issues.redhat.com/browse/MGDOBR-872

This PR depends on:

- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/863

I'm adding support for the new format of masked values (empty object, `{}`) inside processors.

To test this you will have to check your browser developer tools console and inspect network requests.

Steps to test it:
- `npm run start:mocked` (until the related PR is merged, you can only use the mocked APIs)
- open a bridge and click on create new processor
- select the sink type
- provide a processor name
- select an action with sensitive fields (like `AWS Lambda`)
- fill up the action configuration parameters
- click on the create processor button
- inspect the payload of the POST request that creates the processor. It will include the sensitive fields values as you provided them
- in the processors list page, wait until the new processor gets `ready` and then click on its name
- as the processor detail page opens, inspect the request to `GET /bridges/{bridgeId}/processors/{processorId}`, its response will report an empty object (`{}`) as value for the sensitive fields
- the sensitive values are displayed as a bunch of asterisks in the properties list
- now click on the edit button
- change an optional param like the transformation template
- notice that the sensitive fields in the form are empty, no value is displayed
- save the processor
- inspect the payload of the PUT request that saves the processor; the secret fields values are empty objects (`{}`), meaning the back-end will not overwrite the existing value provided when creating the processor
- finally, go back to the processors list and wait until the processor becomes `ready`
- click on the processor name to open its detail
- click on edit
- change one of the secret parameters by adding a new value
- save the processor
- inspect the payload of the PUT request that saves the processor; the edited secret parameters will have the new value, meaning the back-end will overwrite the previous one.
